api/models/score.py

```diff
@@ -403,12 +403,19 @@ def getDynaboardByTask(
             data_list.append(
                 {
                     "model_id": model.id,
-                    "model_name": model.name if model.is_published else None,  # Don't
-                    # give away the users for unpublished models.
-                    "uid": model.uid if model.is_published else None,
-                    "username": uid_to_username[model.uid]
-                    if model.is_published
-                    else None,
+                    "model_name": model.name if model.is_published else None,
+                    # Don't give away the users for unpublished models.
+                    "uid": None
+                    if not model.is_published
+                    else -1
```

Comment:
I've done this so that there isn't any additional overhead on the frontend when figuring out the "viewability" of a model. Specifically for the DynaBoard:

- published and public (actual `uid`, actual `username`)
- published but anonymous (`-1`, `"anonymous"`)
- not published (`None`, `None`)

This is so I can just do things like this https://github.com/facebookresearch/dynabench/pull/780#discussion_r729264973 (which I'll migrate to `uid` as you suggested below) instead of re-figuring out the status of each model on the frontend (which could get even more messy on things like ModelPage, where there are 5 possible "viewability" states).

However I'm open to doing that and dropping the `-1` and `anonymous` if you think it makes the code clearer.

Reply:
Yeah I think that would make the code cleaner. My understanding is that we can keep the uid none both in the scenario where: 1) it is published and anonymous and 2) not published. In both places, it is anonymous.